src/core/evaluator.js

```diff
@@ -1125,30 +1129,16 @@ var PartialEvaluator = (function PartialEvaluatorClosure() {
               var offset;
               for (var j = 0, jj = items.length; j < jj; j++) {
                 if (typeof items[j] === 'string') {
-                  buildTextGeometry(items[j], textChunk);
-                } else {
-                  var val = items[j] / 1000;
-                  if (!textState.font.vertical) {
-                    offset = -val * textState.fontSize * textState.textHScale *
-                      textState.textMatrix[0];
-                    textState.translateTextMatrix(offset, 0);
-                    textChunk.width += offset;
-                  } else {
-                    offset = -val * textState.fontSize *
-                      textState.textMatrix[3];
-                    textState.translateTextMatrix(0, offset);
-                    textChunk.height += offset;
-                  }
-                  if (items[j] < 0 && textState.font.spaceWidth > 0) {
-                    var fakeSpaces = -items[j] / textState.font.spaceWidth;
-                    if (fakeSpaces > MULTI_SPACE_FACTOR) {
-                      fakeSpaces = Math.round(fakeSpaces);
-                      while (fakeSpaces--) {
-                        textChunk.str.push(' ');
-                      }
-                    } else if (fakeSpaces > SPACE_FACTOR) {
+                  buildTextGeometry(items[j], textChunk, items[j - 1]);
```

Comment:
When `j === 0`, this will result in `j - 1` being out of range of the array. As we've seen before, e.g. in https://github.com/mozilla/pdf.js/issues/5663#issuecomment-70983080, that can slow things down in some browsers. Hence I'm wondering if it makes sense to try and avoid that, e.g. do this instead:
```javaScript
buildTextGeometry(items[j], textChunk, (j > 0 ? items[j - 1] : 0));
```